docs/src/conceptual/solution-selection-catalogs.rst

```diff
@@ -0,0 +1,189 @@
+    - library:
+        map:
+          Contraction_l_Alik_Bjlk_Cijk_Dijk:                 # [B_]
+            ...
+            rows:                                            # [C_]
+            - library: {type: Placeholder, value: TensileLibrary_Type_SS_..._fallback}
+              predicate:
+                type: And
+                value:
+                - type: TypesEqual
+                  value: [Float, Float, Float, Float]
+                - {type: HighPrecisionAccumulate, value: false}
+                - {type: F32XdlMathOp, value: Float}
+                - {type: StochasticRounding, value: false}
+            - ...
+            type: Problem
+            ...
+          Contraction_l_Alik_Bljk_Cijk_Dijk:
+            rows:
+              - ...
+            type: Problem                                    # [_C]
+        property: {type: OperationIdentifier}
+        type: ProblemMap                                     # [_B]
+      predicate: {type: TruePred}
+    type: Hardware                                           # [_A]
+  solutions: []
+
+Line **[A]** shows the top level of the parent catalog, which contains a single row for each hardware architecture.
+Line **[B]** shows the problem map for the operation *Contraction_l_Alik_Bjlk_Cijk_Dijk*.
+Line **[C]** shows the problem type and predicates used to match against exact solutions contained in the child catalogs.
+
+Mode 2: Merge files
+-------------------
+
+.. warning::
+    This feature is not recommended and is in the process of being deprecated.
+
+When ``--merge-files`` is enabled, one solution catalog is generated for each architecture, named
+
+.. centered:: TensileLibrary_<gfx>.yaml
+
+The catalog contains information about supported GEMM types and 
+solution metadata that is used to locate the optimal kernel for a requested GEMM. This pattern
+has the drawback that all code object libraries are loaded eagerly,
+thereby increasing both the initialization time and memory footprint of the calling application.
+
+**Example**
+
+Say you're building libraries for gfx908 and gfx90a with ``--merge-files``. The build output directory would look like this
+
+.. code-block:: bash
+
+    build/
+    └── library/
+        ├── Kernels.so-000-gfx1030.hsaco
+        ├── Kernels.so-000-gfx1030.hsaco
+        ├── Kernels.so-000-gfx1030.hsaco
+        ├── Kernels.so-000-gfx900.hsaco
+        ├── Kernels.so-000-gfx906.hsaco
+        ├── TensileLibrary_gfx1030.co
+        ├── TensileLibrary_gfx1030.yaml
+        ├── TensileLibrary_gfx900.co
+        ├── TensileLibrary_gfx900.yaml
+        ├── TensileLibrary_gfx906.co
+        └── TensileLibrary_gfx906.yaml
+
+--------------------
+
+.. [1] Previously these files were called *master solution libraries* because they contain two top level keys, "solutions" and "library". The term *solution selection catalog* was later adopted to clarify the purpose of this file within the larger context of the Tensile C++ API.
```

Comment:
```suggestion
.. [1] Previously these files were named *master solution libraries* because they consisted of two top-level keys, "solutions" and "library". The term *solution selection catalog* was later adopted to clarify the purpose of this file within the larger context of the Tensile C++ API.
```